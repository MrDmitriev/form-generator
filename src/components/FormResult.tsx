import { FormItem } from '../types/FormItem';
import { parseFormConfigToJSON } from '../utils/common';
import FormElement from './FormElement';

type Props = {
	formConfig: string;
}

export default function FormResult({formConfig}: Props) {
	const parsedFormsConfig = parseFormConfigToJSON(formConfig)	;
	const { items = [], buttons = [] } = parsedFormsConfig;
	return (
		<div id="result" className="tabcontent active">
			{items.map((element: FormItem) => <FormElement element={element} />)}

			<div className='form-result__buttons'>
				{buttons.map((button: React.ReactNode) => <button className='form-button'>{button}</button>)}
			</div>

		</div>
	)
}
