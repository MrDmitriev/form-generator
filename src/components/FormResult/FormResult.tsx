import { FormItem } from '../../types/FormItem';
import { parseFormConfigToJSON } from '../../utils/common';
import FormElement from '../FormElement/FormElement';

type Props = {
	formConfig: string;
}

//Renders form, based on provided form configuration
export default function FormResult({formConfig}: Props) {
	const parsedFormsConfig = parseFormConfigToJSON(formConfig);
	const { items = [], buttons = [] } = parsedFormsConfig;

	const emptyFormConfigWarning = <div className="tabcontent">You did not provide any valid form configuration</div>
	return parsedFormsConfig ? (
		<div id="result" className="tabcontent active">
			{items.map((element: FormItem) => <FormElement element={element} />)}

			<div className='form-result__buttons'>
				{buttons.map((button: React.ReactNode) => <button className='form-button'>{button}</button>)}
			</div>

		</div>
	) : emptyFormConfigWarning
}
