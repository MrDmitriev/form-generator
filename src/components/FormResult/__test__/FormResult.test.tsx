import renderer from 'react-test-renderer';
import FormResult from '../FormResult';
import mockFormConfig from '../../../mock-content/form-config-test.json';

describe("FormResult component", () => {
  it(`should match snapshot`, () => {
		const props = {
			formConfig: JSON.stringify(mockFormConfig)
		}
	
		const wrapper = renderer.create(<FormResult {...props} />).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});