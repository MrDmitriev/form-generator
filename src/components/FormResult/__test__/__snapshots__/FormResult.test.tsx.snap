// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormResult component should match snapshot 1`] = `
<div
  className="tabcontent active"
  id="result"
>
  <div
    className="form-element"
  >
    <label
      className="form-element__label"
      htmlFor="numberfield"
    >
      Count
    </label>
    <input
      className="form-element__input"
      id="numberfield"
      type="number"
    />
  </div>
  <div
    className="form-element"
  >
    <label
      className="form-element__label"
      htmlFor="textfield"
    >
      Caption
    </label>
    <input
      className="form-element__input"
      id="textfield"
      type="text"
    />
  </div>
  <div
    className="form-element"
  >
    <label
      className="form-element__label"
      htmlFor="textarea"
    >
      Long text
    </label>
    <textarea
      className="form-element__input"
      id="textarea"
      rows={5}
    />
  </div>
  <div
    className="form-element"
  >
    <label
      className="form-element__label"
      htmlFor="checkbox"
    >
      Is Editable
    </label>
    <input
      id="checkbox"
      type="checkbox"
    />
  </div>
  <div
    className="form-element"
  >
    <label
      className="form-element__label"
      htmlFor="datefield"
    >
      Date
    </label>
    <input
      className="form-element__input"
      id="datefield"
      type="date"
    />
  </div>
  <div
    className="form-element"
  >
    <label
      className="form-element__label"
      htmlFor="radiobutton"
    >
      Radio
    </label>
    <input
      id="radiobutton"
      type="radio"
    />
  </div>
  <div
    className="form-result__buttons"
  >
    <button
      className="form-button"
    >
      Ok
    </button>
    <button
      className="form-button"
    >
      Cancel
    </button>
  </div>
</div>
`;
